src/Microsoft.PowerShell.Commands.Utility/commands/utility/UnblockFile.cs

```diff
@@ -128,6 +135,30 @@ protected override void ProcessRecord()
                     }
                 }
             }
+#else
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                string errorMessage = UnblockFileStrings.LinuxNotSupported;
+                Exception e = new NotImplementedException(errorMessage);
+                ThrowTerminatingError(new ErrorRecord(e, "LinuxNotSupported", ErrorCategory.NotImplemented, null));
+                return;
+            }
+
+            foreach (string path in pathsToProcess)
+            {
+                if(IsBlocked(path))
+                {
+                    UInt32 result = RemoveXattr(path,MacBlockAttribute,RemovexattrFollowSymLink);
+                    if(result != 0)
+                    {
+                        string errorMessage = string.Format(CultureInfo.CurrentUICulture, UnblockFileStrings.UnblockError, path);
+                        Exception e = new InvalidOperationException(errorMessage);
```

Comment:
maybe this should be `System.PlatformNotSupportedException`

Reply:
definitely not, we are on a supported platform here.

Reply:
maybe you meant line 142... changed it there.